### PR TITLE
Add templates for deploying a controller and workers.

### PR DESCRIPTION
**Pull request by chrisgavin**
This adds the Azure Resource Manager templates for deploying LGTM Enterprise to Azure.

The templates have a `$version` variable that will be replaced automatically when the images get published but should otherwise work as-is.

### CODE REVIEW

File: controller.json [hunk 1]
```diff
@@ -0,0 +1,564 @@
+{
+	"$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+	"contentVersion": "0.0.0.0",
+	"parameters": {
+		"identifier": {
+			"type": "string",
+			"minLength": 1,
+			"maxLength": 16,
```
Comment (rneatherway) [inline controller.json:8]: Why limit this to 16 characters?

File: controller.json [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+				"Standard_A11",
+				"Standard_E48d_v4",
+				"Standard_E64d_v4",
+				"Standard_E48ds_v4",
+				"Standard_E64-16ds_v4",
+				"Standard_E64ds_v4",
+				"Standard_ND40rs_v2",
+				"Standard_NV8as_v4",
+				"Standard_NV16as_v4",
+				"Standard_NV32as_v4"
```
Comment (rneatherway) [inline controller.json:273]: Very exhaustive ;-)

File: controller.json [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+								"id": "[resourceId(resourceGroup().name, 'Microsoft.Network/publicIpAddresses', variables('publicIPName'))]"
+							}
+						}
+					}
+				]
+			}
+		},
+		{
+			"type": "Microsoft.Compute/disks",
+			"apiVersion": "2019-11-01",
```
Comment (rneatherway) [inline controller.json:500]: Interesting date-based versioning

**Review by rneatherway — APPROVED** [review]
I'm not 100% clear on the testing situation for this one, but as long as Alistair's happy and we cover it during Testing Day I'm fine with it.

File: controller.json (hunk 1 above)
Comment (chrisgavin) [reply controller.json:8]: The identifier gets appended to other strings to form resource names and those resource names have varying length limits. 16 seemed both long enough to offer an easily identifiable name and short enough to still be safe when appended to other strings and used as resource names. We can probably make it longer than 16 if you think a higher limit is worthwhile.

Comment (rneatherway) [reply controller.json:8]: No that's good enough justification for me!

**Pull request merged**